tests/avocado/avocado_qemu/test.py

```diff
@@ -369,35 +384,11 @@ def __init__(self, methodName=None, name=None, params=None,
         if machine:
             self.vm.args.extend(['-machine', machine])
 
-    def request_image(self, path=None, snapshot=None, extra=None):
-        """
-        Add image to the `self.vm` using params or arguments.
-
-        Unless it's overridden by arguments it uses following test params
-        to specify the image:
-
-        * image_path - defines the path to the user-image. If not specified
-                       it uses "QEMU_ROOT/boot_image_$arch.qcow2"
-        * image_snapshot - whether to use "snapshot=on" (snapshot=off is not
-                           supplied)
-        * image_extra - free-form string to extend the "-drive" params
-
-        :param path: Override the path ("image_path" param is used otherwise)
-        :param snapshot: Override the usage of snapshot
-        :param extra: Extra arguments to be added to drive definition
-        """
-        if snapshot is None:
-            snapshot = self.params.get("image_snapshot", default=True)
-        if extra is None:
-            extra = self.params.get("image_extra", default="")
-        if path is None:
-            path = self.params.get("image_path")
-            if path is None:
-                arch = self.vm.arch
-                path = os.path.join(QEMU_ROOT, "boot_image_%s.qcow2" % arch)
-        if not os.path.exists(path):
-            self.error("Require a bootable image, which was not found. "
-                       "Please provide one in '%s'." % path)
-        if snapshot:
-            extra += ",snapshot=on"
-        self.vm.args.extend(['-drive', 'file=%s%s' % (path, extra)])
+        image_path = self.params.get("image_path", default=None)
+        if image_path is not None:
```

Comment:
The reason I added the `request_image` was that I expect people to run multiple tests with the same params and only tests that require the image should specify it. Adding the image always when the `image_path` exists would make all the tests to define the image, which would result in different environments when between different executions, which is definitely not anything I'd support.

Reply:
btw that is also why I specified the default, where `QEMU_ROOT/boot_image_$arch.qcow2` is used when image is not specified by params, because I don't see myself always including the params. Simpler for me is to create symlink in `QEMU_ROOT`, set the user/pass and be done with it.